etk-asm/src/asm.rs

```diff
@@ -394,26 +421,15 @@ impl Assembler {
                 Ok(())
             }
             RawOp::Op(AbstractOp::MacroDefinition(_)) => Ok(()),
-            RawOp::Op(AbstractOp::Macro(ref m)) => {
-                match self.declared_macros.get(&m.name) {
-                    // Do nothing if the instruction macro has been defined.
-                    Some(MacroDefinition::Instruction(_)) => (),
-                    _ => {
-                        assert_eq!(self.pending_len, Some(0));
-                        self.pending_len = None;
-                        self.pending.push_back(rop);
-                    }
-                }
-                Ok(())
-            }
+            RawOp::Op(AbstractOp::Macro(_)) => Ok(()),
```

Comment:
If I understand this correctly, hitting a macro invocation inside this function is a bug, right? We should either merge `push` and `push_rawop`, or put `unreachable!()` here instead of `Ok(())`.

Reply:
Not really. While they have all been processed at this point, they are not removed from the original code. So this is simply to ignore them.

Reply:
Yes, it's not exactly a bug, but since the macro had already been processed before, it was simply ignored. I will combine both pushes to simplify the code.